starky/src/lookup.rs

```diff
@@ -719,9 +722,12 @@ pub(crate) fn eval_helper_columns_circuit<F: RichField + Extendable<D>, const D:
     consumer: &mut RecursiveConstraintConsumer<F, D>,
 ) {
     if !helper_columns.is_empty() {
-        for (j, chunk) in columns.chunks(constraint_degree - 1).enumerate() {
-            let fs =
-                &filter[(constraint_degree - 1) * j..(constraint_degree - 1) * j + chunk.len()];
+        for (j, chunk) in columns
+            .chunks(constraint_degree.checked_sub(1).unwrap_or(1))
+            .enumerate()
+        {
+            let fs = &filter[(constraint_degree.saturating_sub(1)) * j
+                ..(constraint_degree.saturating_sub(1)) * j + chunk.len()];
```

Comment:
Shouldn't this be as above? 
```suggestion
let chunk_size = constraint_degree.checked_sub(1).unwrap_or(1);
        for (chunk, (fs, &h)) in columns
            .chunks(chunk_size)
            .zip(filter.chunks(chunk_size).zip(helper_columns))
        {
```
Especially since, as mentioned in Robin's comments, `saturating_sub` would yield 0 but I think we need 1?